lib/src/main/java/io/github/taz/java/instagram/api/IgClient.java

```diff
@@ -53,32 +57,34 @@ public String getCookies() {
 
     public void login() {
         httpClient.sendAsync(new QeSyncRequest().formRequest(this), BodyHandlers.discarding())
-            .thenAccept(response -> {
-                HttpHeaders headers = response.headers();
-                String encryptionId = headers.firstValue("ig-set-password-encryption-key-id").get();
-                String encryptionKey = headers.firstValue("ig-set-password-encryption-pub-key").get();
-
-                try {
-                    String encryptedPassword = encryptPassword(password, encryptionId, encryptionKey);
-                    sendRequest(new AccountsLoginRequest(username, encryptedPassword)).join();
-                } catch (Exception e) {
-                    e.printStackTrace();
-                }
-            })
-            .join();
+                .thenAccept(response -> {
+                    logger.info("Response received from {}", response.uri());
+                    HttpHeaders headers = response.headers();
+                    String encryptionId = headers.firstValue("ig-set-password-encryption-key-id").get();
+                    String encryptionKey = headers.firstValue("ig-set-password-encryption-pub-key").get();
+
+                    try {
+                        String encryptedPassword = encryptPassword(password, encryptionId, encryptionKey);
+                        sendRequest(new AccountsLoginRequest(username, encryptedPassword)).join();
+                    } catch (Exception e) {
+                        logger.error("Error occurred while encrypting password or sending request", e);
```

Comment:
actually sendRequest doesn't throw any exception so its def the encryption, also tell that its invalid password, coz thats the only case in which it can throw error